### PR TITLE
[TF][Op] Op where

### PR DESCRIPTION
**Pull request by wweic**
@yongwww  @srkreddy1238 @zhiics Please take a look.

### CODE REVIEW

File: tests/python/frontend/tensorflow/test_forward.py [hunk 1]
```diff
@@ -63,24 +89,32 @@ def run_tvm_graph(graph_def, input_data, input_node, num_output=1,
...
-    tvm_output_list = [m.get_output(i).asnumpy() for i in range(num_output)]
-    return tvm_output_list
+    if mode == 'interp':
+        ex = relay.create_executor("debug", mod=mod, ctx=tvm.cpu(), target="llvm")
+        inputs = []
+        for param in mod['main'].params:
+            inputs.append(tvm.nd.array(params[param.name_hint]))
+        result = ex.evaluate()(*inputs)
+        return vmobj_to_list(result)
+    else:
```
Comment (yongwww) [inline tests/python/frontend/tensorflow/test_forward.py]: could you please help add `mode =='vm'` here?

File: tests/python/frontend/tensorflow/test_forward.py [hunk 2]
```diff
@@ -325,6 +359,19 @@ def test_forward_biasadd():
     _test_biasadd([4, 17, 17, 19], 'NHWC')
     _test_biasadd([4, 3, 3, 124], 'NHWC')
 
+def _test_forward_where(input_shape):
+    with tf.Graph().as_default():
+        dtype =  tf.float32
+        t = tf.constant(np.random.choice([0, 1, 2, 3], size=input_shape).astype(dtype.name))
```
Comment (yongwww) [inline tests/python/frontend/tensorflow/test_forward.py]:
```suggestion
        t = tf.constant(np.random.choice([0, 1, -2, 3, -1, 0.1, -0.2], size=input_shape).astype(dtype.name))
```

File: tests/python/frontend/tensorflow/test_forward.py [hunk 3]
```diff
@@ -46,8 +46,34 @@ def convert_to_list(x):
...
+        return [o.asnumpy().tolist()]
+    elif isinstance(o, tvm.relay.backend.vmobj.DatatypeObject):
+        result = []
+        for f in o:
+            result.extend(vmobj_to_list(f))
+        return result
+    elif isinstance(o, tvm.relay.backend.interpreter.TupleValue):
+        result = []
+        for f in o.fields:
+            result.append(vmobj_to_list(f))
```
Comment (yongwww) [inline tests/python/frontend/tensorflow/test_forward.py]: so result here is list of list?

File: tests/python/frontend/tensorflow/test_forward.py [hunk 4]
```diff
@@ -325,6 +359,19 @@ def test_forward_biasadd():
...
 
+def _test_forward_where(input_shape):
+    with tf.Graph().as_default():
+        dtype =  tf.float32
+        t = tf.constant(np.random.choice([0, 1, 2, 3], size=input_shape).astype(dtype.name))
+        out = tf.where(t)
+        compare_tf_with_tvm([], [], out.name, mode='interp')
+
+def test_forward_argwhere():
+    _test_forward_where((5, 5))
```
Comment (yongwww) [inline tests/python/frontend/tensorflow/test_forward.py]: \+ `_test_forward_where((5,))`

File: tests/python/frontend/tensorflow/test_forward.py (hunk 3 above)
Comment (wweic) [reply tests/python/frontend/tensorflow/test_forward.py]: yes

File: tests/python/frontend/tensorflow/test_forward.py (hunk 1 above)
Comment (wweic) [reply tests/python/frontend/tensorflow/test_forward.py]: done

**Comment by wweic** [comment]
@icemelon9 could you also help review and merge it?

**Comment by icemelon9** [comment]
Thanks @wweic

**Pull request merged**